SDL/SDLJoystick.cpp

```diff
@@ -125,7 +132,38 @@ void SDLJoystick::ProcessInput(SDL_Event &event){
 #endif
 			break;
 		}
+
+	case SDL_JOYDEVICEADDED:
+		{
+			int deviceIndex = event.jdevice.which;
+			if (deviceIndex >= joys.size()) {
+				joys.resize(deviceIndex+1);
+			}
+			joys[deviceIndex] = SDL_JoystickOpen(deviceIndex);
+			SDL_JoystickEventState(SDL_ENABLE);
+			break;
+		}
+
+	case SDL_JOYDEVICEREMOVED:
+		{
+			int deviceIndex = getDeviceIndex(event.jdevice.which);
+			if (deviceIndex >= 0){
+				SDL_JoystickClose(joys[deviceIndex]);
+				joys[deviceIndex] = 0;
+			}
+			break;
+		}
+	}
+}
+
+int SDLJoystick::getDeviceIndex(int instanceId){
+	for (int i = 0; i < joys.size(); i++){
+		SDL_Joystick *joy = joys[i];
+		if (SDL_JoystickInstanceID(joy) == instanceId){
```

Comment:
How's the performance of this?  People often complain about latency, just wondering if we ought to cache this in the array (or a separate one) instead.

Also, please use a space between parens and curlies - `) {`.

-[Unknown]